app/models/concerns/taxonomix.rb

```diff
@@ -116,13 +111,27 @@ def inner_select(taxonomy, inner_method = which_ancestry_method)
     def admin_ids
       User.unscoped.where(:admin => true).pluck(:id) if self == User
     end
+
+    def scope_by_taxable_ids(scope)
+      case taxable_ids
+      when nil
+        scope
+      when []
+        # If *no* taxable ids were found, then don't show any resources
+        scope.where(:id => [])
+      else
+        # We need to generate the WHERE part of the SQL query as a string,
+        # otherwise the default scope would set id on each new instance
+        # and the same taxable_id on taxable_taxonomy objects
+        scope.where("#{self.table_name}.id IN (#{taxable_ids.join(',')})")
```

Comment:
`taxable_ids` should remain cached to prevent calling this expensive function twice